service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/model/service/Profile.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.servicespec.commons.model.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.utils.ObjectMapperFactory;
+import io.hyscale.servicespec.commons.exception.ServiceSpecErrorCodes;
+import io.hyscale.servicespec.commons.json.parser.JsonTreeParser;
+
+import java.io.IOException;
+
+public class Profile implements HyscaleFileInputProcessor {
+    private JsonNode root;
+
+    public Profile(JsonNode root) {
+        this.root = root;
+    }
+
+    public Profile(String profilePath) throws HyscaleException {
+        ObjectMapper mapper = ObjectMapperFactory.yamlMapper();
+        try {
+            this.root = mapper.readTree(profilePath);
+        } catch (IOException e) {
+            throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_PROFILE_PARSE_ERROR);
+        }
+    }
+
+    @Override
+    public JsonNode get(String path) {
```

Comment:
These method functionality is common in service spec as well as profile, should we look for a way to remove the duplication?

Reply:
The duplication should be fine for now, need to think of a way when we move to object model with custom parameters. This might affect then